lib/SPIRV/CMakeLists.txt

```diff
@@ -39,6 +39,7 @@ add_llvm_library(LLVMSPIRVLib
   LINK_COMPONENTS
     Analysis
     BitWriter
+    CodeGen
```

Comment:
The tradeoff here is having an extra and quite heavy LLVM library linked to llvm-spirv vs having(and maintaining) extra source code which is a copy-paste of a function from LLVM CodeGen lib.
The source we would copy is [this function](https://github.com/KornevNikita/llvm/blob/a8bb540ac7d683cdf087a0676ffdef9d3a4f7d2d/llvm/lib/CodeGen/IntrinsicLowering.cpp#L53-L145), ~100 lines.
@AlexeySachkov, @MrSidims what do you think?

Reply:
libLLVMCodeGen is indeed the biggest component of LLVM.  Perhaps we can propose to move the intrinsic lowering elsewhere (e.g. into TransformUtils)?

For us, linking against libLLVMCodeGen shouldn't be an issue, so that would still have our preference over duplicating code.